keras_nlp/src/utils/transformers/safetensor_utils.py

```diff
@@ -42,12 +42,42 @@ def __init__(self, preset):
         else:
             self.safetensor_config = None
         self.safetensor_files = {}
+        self.prefix = None
+
+    def get_prefix_from_keys(self, hf_weight_key, hf_config_keys):
+        """
+        Determine and return the prefix for a given hf weight key.
+
+        This method checks if there's a common prefix for the weight keys and caches it
+        for future use.
+
+        Args:
+            hf_weight_key (str): The hf weight key to check for a prefix.
+            hf_config_keys (iterable): An iterable of all hf weight keys to check against.
+
+        Returns:
+            str: The full key including the prefix (if any).
+        """
+        if self.prefix is None:
+            for full_key in hf_config_keys:
+                if (
+                    full_key.endswith(hf_weight_key)
+                    and full_key != hf_weight_key
+                ):
+                    self.prefix = full_key[: -len(hf_weight_key)]
+                    break
+            else:
+                self.prefix = ""
+
+        return self.prefix + hf_weight_key
```

Comment:
maybe flip this to an early return a the top? Save us needing to have triple indented control blocks in this fn.

```
if self.prefix is not None:
    # Re-use the cached prefix if we found one before.
    return self.prefix + hf_weight_key
```

Reply:
Optimized it. Thank you for the suggestion.